### DISCUSSION TITLE
Read only fields

### DISCUSSION THREAD

**Discussion by SueSpevack** [UKHomeOffice/design-system#503] [Ideas]
From recent design discussions in the community on the designs of non editable fields, there is a need for teams to have a clearer understanding on best practise on the application of read only fields.

We tested in recent research sessions (on an internal service), using greyed out fields to indicate that some fields were not editable. Users found it difficult to distinguish the differences on the screen with fields that were editable, and the colour contrast on the text input doesn't provide enough contrast when using the Home Office design system. 

Other teams have removed the input itself and just showed the saved content, but this needs further research and insights.

There are also questions in regards to accessibility, and how users can understand what they can/can't edit.

There is a need for further research and guidance to support designers making informed decisions on the best way to implement read only fields.

**Comment by guermanc** [comment]
There are a couple of discussions on this with multiple reasons why we shouldn't be using them that agree with all the points that you raised at https://github.com/alphagov/govuk_elements/issues/374 and https://github.com/alphagov/govuk-frontend/pull/3187

As you have suggested I would remove the field and just present the contents as text, which would make the information more accessible. Users will know that the text is not editable as it won't be in an input field.

There may be a need for disabled input fields, but as noted in https://design-system.service.gov.uk/components/button/#disabled-buttons only if it makes the user interface easier to understand, and so will vary on a case-by-case basis and require additional usability research.

**Comment by benjystanton** [comment]
Interested in this too. Some clear guidance would be useful! 

These blog posts offer some useful background too

- https://adrianroselli.com/2024/02/dont-disable-form-controls.html
- https://kittygiraudel.com/2024/03/29/on-disabled-and-aria-disabled-attributes/
